pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -396,6 +396,14 @@ func (m *Multicluster) deleteCluster(clusterID cluster.ID) {
 		log.Infof("cluster %s does not exist, maybe caused by invalid kubeconfig", clusterID)
 		return
 	}
+
+	// Unregister networks handler
+	if kc.opts.MeshNetworksWatcher != nil {
+		kc.opts.MeshNetworksWatcher.DeleteNetworksHandler(kc.networksHandler)
+	}
+	// Unregister mesh handler
+	kc.unregisterMeshWatcherHandler()
```

Comment:
Suggest group register and unregister together, in case we add a new watcher later, we miss clean it up here.

So i think we should move to kube Controller.CleanUp

Reply:
Good suggestion.